pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -623,8 +691,132 @@ func (r *ReconcileSyncSetInstance) applySyncSetPatches(ssi *hivev1.SyncSetInstan
 	return nil
 }
 
-func (r *ReconcileSyncSetInstance) reconcileDeletedSyncSetResources(applyMode hivev1.SyncSetResourceApplyMode, dynamicClient dynamic.Interface, existingStatusList, newStatusList []hivev1.SyncStatus, err error, ssiLog log.FieldLogger) ([]hivev1.SyncStatus, error) {
-	ssiLog.Debugf("reconciling syncset resources, existing: %d, actual: %d", len(existingStatusList), len(newStatusList))
+// applySyncSetSecretReferences
+func (r *ReconcileSyncSetInstance) applySyncSetSecretReferences(ssi *hivev1.SyncSetInstance, secretReferences []hivev1.SecretReference, dynamicClient dynamic.Interface, h Applier, ssiLog log.FieldLogger) error {
+	syncStatusList := []hivev1.SyncStatus{}
+
+	var applyErr error
+	for _, secretReference := range secretReferences {
+		secret := &corev1.Secret{}
+		err := r.Get(context.Background(), types.NamespacedName{Name: secretReference.Source.Name, Namespace: secretReference.Source.Namespace}, secret)
+		if err != nil {
+			ssiLog.WithError(err).WithField("secret", fmt.Sprintf("%s/%s", secretReference.Source.Name, secretReference.Source.Namespace)).Error("cannot read secret")
+			return err
+		}
+
+		hash, err := controllerutils.GetChecksumOfObject(secret)
+		if err != nil {
+			ssiLog.WithError(err).WithField("secret", fmt.Sprintf("%s/%s", secretReference.Source.Name, secretReference.Source.Namespace)).Error("unable to compute secret hash")
+			return err
+		}
+
+		secretReferenceSyncStatus := hivev1.SyncStatus{
+			APIVersion: secretReference.Target.APIVersion,
+			Kind:       secretReference.Target.Kind,
+			Name:       secretReference.Target.Name,
+			Namespace:  secretReference.Target.Namespace,
+			Resource:   secret.GetObjectMeta().GetResourceVersion(),
+			Hash:       hash,
+		}
+
+		var secretReferenceSyncConditions []hivev1.SyncCondition
+
+		// determine if secret is found, different or should be reapplied based on last probe time
+		found := false
+		different := false
+		shouldReApply := false
+		for _, rss := range ssi.Status.SecretReferences {
```

Comment:
Can we make the following function instead of having this for loop and subsequent if statement here and in `reconcileDeletedSyncSetResources`?
```
func findSecretRefStatus(hivev1.SyncStatus, []hivev1.SyncStatus) *hivev1.SyncStatus
```